### PR TITLE
Support for geospatial types - point, line, box, polygon etc

### LINKED ISSUE #282: Support for geospatial types - point, line, box, polygon etc
## Feature Request
Please don't mind as I'm a beginner wrt requesting OSS features.
I would like to have support for geo-spatial types like point, line, polygon etc.

#### Is your feature request related to a problem? Please describe
Our current project is JDBC based and there we use postgis-jdbc to save our data along with geospatial data in postgresql. we save mostly coordinates (Points) or lines or polygons for some markings done on maps. We were wondering if we can use r2dbc, if it had the support for these data types.

#### Describe the solution you'd like
I would like to have support for geospatial types of postgresql.

#### Geometry type support

- [x] `point` (#283)
- [ ] `line`
- [ ] `lseg`
- [ ] `box`
- [ ] `path` (open, closed)
- [ ] `polygon`
- [x] `circle` (#300)

### PR DESCRIPTION
**Pull request by anshlykov**
Resolves #282

### CODE REVIEW

**Comment by mp911de** [comment]
Thanks for your contribution. Please note there's another pull request (#305) that ships with `lseg` support. Since this PR covers the remaining geo codecs it makes sense to consider this one over #305.

File: src/main/java/io/r2dbc/postgresql/codec/Line.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import java.util.Objects;
+
+/**
+ * Value object that maps to the {@code line} datatype in Postgres.
+ * <p>
+ * Uses {@code double} to represent the coordinates.
+ */
+public final class Line {
+
+    private final double a;
```
Comment (mp911de) [inline src/main/java/io/r2dbc/postgresql/codec/Line.java:12]: We should add a bit of Javadoc to the fields.

File: src/main/java/io/r2dbc/postgresql/codec/Line.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    private final double c;
+
+    private Line(double a, double b, double c) {
+        this.a = a;
+        this.b = b;
+        this.c = c;
+    }
+
+    public static Line of(double a, double b, double c) {
```
Comment (mp911de) [inline src/main/java/io/r2dbc/postgresql/codec/Line.java:24]: Would it make sense to add factory methods accepting two points or even x1/y1, x2/y2?

**Comment by anshlykov** [comment]
@mp911de thanks for the review. I updated the `Line` class

**Comment by mp911de** [comment]
Thank you for your contribution. That's merged, polished, and backported now.

**Pull request closed without merging**